allennlp/data/semparse/action_space_walker.py

```diff
@@ -36,19 +36,34 @@ def _walk(self) -> None:
 
         self._completed_paths = []
         actions = self._world.get_valid_actions()
+        # Overview: We keep track of the buffer of non-terminals to expand, and the action history
+        # for each incomplete path. At every iteration in the while loop below, we iterate over all
+        # incomplete paths, expand one non-terminal from the buffer in a depth-first fashion, get
+        # all possible next actions triggered by that non-terminal and add to the paths. Then, we
+        # check whether the expanded paths, to see if they are 1) complete, in which case they are
```

Comment:
Grammar in this sentence is off.  Probably remove "whether".

Reply:
Done.